schemas/event.yaml

```diff
@@ -0,0 +1,146 @@
+$schema: http://json-schema.org/draft-07/schema#
+title: Event
+properties:
+  event_id:
+    description: UUID for the event, will be generated upon ingestion if not specified.
+    type: string
+    format: uuid
+    required: false
+  event_source:
+    description: Identifier for the system that captured the event.
+    type: string
+    required: true
+  event_type:
+    description: Source-specific identifier for the type of event (e.g. Snapshot, Delete). This allows
+      the source to encode a special semantic for the event.
+    type: string
+    required: false
+  account_number:
+    description: Account identifier for the relevant account.
+    type: string
+    required: true
+  service_type:
+    description: Identifier for the type of service being measured. (E.g. RHEL System, OpenShift Cluster)
+    type: string
+    required: true
+  instance_id:
+    description: Primary identifier for the instance of the service, according to the event producer.
+    type: string
+    required: true
+  timestamp:
+    description: Time that the event happened or the snapshot was captured.
+    type: string
+    format: date-time
+    required: true
+  expiration:
+    description: End of the range for the event (exclusive).
+    type: string
+    existingJavaType: java.util.Optional<java.time.OffsetDateTime>
+    format: date-time
+    required: false
+  display_name:
+    description: User-set name for the subject.
+    type: string
+    existingJavaType: java.util.Optional<String>
+    required: false
+  inventory_id:
+    description: Host-based inventory ID for the service, if applicable and known.
+    type: string
+    existingJavaType: java.util.Optional<String>
+    required: false
+  insights_id:
+    description: Red Hat Insights ID for the service, if applicable and known.
+    type: string
+    existingJavaType: java.util.Optional<String>
+    required: false
+  subscription_manager_id:
+    description: subscription-manager ID for the service, if applicable and known.
+    type: string
+    existingJavaType: java.util.Optional<String>
+    required: false
+  correlation_ids:
+    description: List of IDs to be used in debugging. Correlation IDs should ideally be unique.
+    type: array
+    items:
+      type: string
+    required: false
+  measurements:
+    description: Captures measurements for the service instance.
+    type: array
+    items:
+      type: object
+      properties:
+        value:
+          description: Measurement value.
+          type: number
+          required: true
+        uom:
+          description: UOM for the measurement. (e.g. Cores, Sockets)
```

Comment:
can this be an enum?

Reply:
@beav It *could*, but then anytime we want to onboard a new UOM, we have to update the schema. Updating a schema is not particular difficult though. :smile:  @lindseyburnett @mstead @awood thoughts?

Reply:
@kahowell
Uom is enumerated (cores/sockets) in our rhsm-subscriptions-api-spec.yaml, not sure if that affects an opinion one way or another.   Are there other UOMs that are currently out there that other applications are using but swatch doesn't support yet?  How often do we anticipate onboarding new UOMs?

In general, I'm leaning towards enum now instead of later.

Reply:
I think we should use an enum now. The intent of this schema is to define the expected data format. Enum here would take the guesswork out.

Reply:
Done